### PR TITLE
Cleanup. Started with JoystickManager (as good as finished)

### PR DESCRIPTION
**Pull request by emielsteerneman**

### CODE REVIEW

File: src/joystick/JoystickManager.cpp [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    init();
+    loop();
+}
+
+/**
+ * Inits SDL and JoystickHandler-map
+ */
+void JoystickManager::init() {
+    std::cout << "[JoystickManager][init] Initializing.." << std::endl;
+    if(SDL_InitSubSystem(SDL_INIT_JOYSTICK))
```
Comment (mrlukasbos) [inline src/joystick/JoystickManager.cpp:22]: this seems like a weird check: if init then show error message?

File: src/joystick/JoystickManager.cpp [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+        std::cout << "SDL_INIT_JOYSTICK failed : " << SDL_GetError() << std::endl;
+    SDL_JoystickEventState(SDL_ENABLE);
+}
+
+/**
+ * Handles events, handles and ticks JoystickHandlers
+ */
+void JoystickManager::loop() {
```
Comment (mrlukasbos) [inline src/joystick/JoystickManager.cpp:30]: function exceeds recommended 20 lines. can be easily split up in smaller functions

File: src/joystick/JoystickManager.cpp [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    int iTicks = 0;
+    int iEvents = 0;
+    SDL_Event event;
+
+    steady_clock::time_point tTickNext = steady_clock::now() + milliseconds(TICK_INTERVAL);
+
+    while(true) {
```
Comment (mrlukasbos) [inline src/joystick/JoystickManager.cpp:37]: you could probably use roboteam_utils::Timer

File: src/joystick/JoystickManager.cpp [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+    std::cout << "[JoystickManager][handleJoystickAdded] Adding joystick " << evt.jdevice.which << std::endl;
+
+    SDL_Joystick* joystick = SDL_JoystickOpen(evt.jdevice.which);
+    if(!joystick){
+        std::cout << "[JoystickManager][handleJoystickAdded] Error! Could not open joystick!" << std::endl;
+        return;
+    }
+
+    int instanceId = SDL_JoystickInstanceID(joystick);
+    Test* t = new Test(instanceId);
```
Comment (mrlukasbos) [inline src/joystick/JoystickManager.cpp:85]: A class called `Test`? what does it do?

File: src/joystick/JoystickManager.cpp [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+ * Takes an SDL_Event and deletes and removes the correct JoystickHandler from the map of JoystickHandlers.
+ */
+void JoystickManager::handleJoystickRemoved(const SDL_Event& evt){
+    std::cout << "[JoystickManager][handleJoystickRemoved] Removing joystick " << evt.jdevice.which << std::endl;
+    delete joystickHandlers.at(evt.jdevice.which);
+    joystickHandlers.erase(evt.jdevice.which);
+    std::cout << "[JoystickManager][handleJoystickAdded] Removed joystick with InstanceID " << evt.jdevice.which << std::endl;
+}
+
+
```
Comment (mrlukasbos) [inline src/joystick/JoystickManager.cpp:100]: redundant newlines

File: src/joystick/JoystickManager.cpp (hunk 4 above)
Comment (mrlukasbos) [reply src/joystick/JoystickManager.cpp:85]: t does not get deleted

File: src/joystick/JoystickManager.cpp [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+    joystickHandlers.insert({instanceId, t});
+    std::cout << "[JoystickManager][handleJoystickAdded] Added joystick with InstanceID " << instanceId << std::endl;
+}
+
+/**
+ * Takes an SDL_Event and deletes and removes the correct JoystickHandler from the map of JoystickHandlers.
+ */
+void JoystickManager::handleJoystickRemoved(const SDL_Event& evt){
+    std::cout << "[JoystickManager][handleJoystickRemoved] Removing joystick " << evt.jdevice.which << std::endl;
+    delete joystickHandlers.at(evt.jdevice.which);
```
Comment (mrlukasbos) [inline src/joystick/JoystickManager.cpp:95]: smart pointers will make your life better

File: src/joystick/JoystickManager.h [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+
+#include <memory>
+#include <map>
+
+#include <SDL.h>
+#include <SDL_joystick.h>
+#include <unistd.h>
+
+/* Simple test class to emulate a JoystickHandler */
+class Test {
```
Comment (mrlukasbos) [inline src/joystick/JoystickManager.h:16]: Why?

**Review by mrlukasbos — COMMENTED** [review]
First short review. The Test class looks very weird to me, but maybe I just don't understand it.

**Comment by emielsteerneman** [comment]
Nevermind, this was not supposed to be pushed to master

**Pull request closed without merging**